### PR TITLE
feat: Option to change color scheme #40

### PR DESCRIPTION
**Pull request by shixin-guo**
[#40]
Users can customize the color of the file. Needs to be a JSON string. it will replace the matching key-value pair of [language-colors.json](https://github.com/githubocto/repo-visualizer/blob/main/src/language-colors.json)

@Wattenberger help review

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -80,6 +80,13 @@ If unspecified, no artifact will be created.
 
 Default: `''` (no artifact)
 
+### `customize_file_colors`
```
Comment (Wattenberger) [inline README.md:83]: let's change this to `file_colors` since it's clear that they will be custom

File: README.md [hunk 2]
```diff
@@ -80,6 +80,13 @@ If unspecified, no artifact will be created.
...
+
+User can customize color of file. need to be a Json String. it will replace the key-value pair of [src/language-colors.json].
```
Comment (Wattenberger) [inline README.md:85]: what about:

> You can customize the colors for specific file extensions. Key/value pairs will extend the [default colors](src/language-colors.json).

File: src/Tree.tsx [hunk 3]
```diff
@@ -32,6 +32,7 @@ type Props = {
   filesChanged: string[];
   maxDepth: number;
   colorEncoding: "type" | "number-of-changes" | "last-change"
+  customizeFileColors?: { [key: string]: string };
```
Comment (Wattenberger) [inline src/Tree.tsx:35]: let's mirror the change above - this should be `fileColors`

**Review by Wattenberger — CHANGES_REQUESTED** [review]
Thanks for putting this PR together! Once we change the naming, we can get this merged in!

File: README.md (hunk 2 above)
Comment (shixin-guo) [reply README.md:85]: Ok thanks very much, My English is not good, I will change it as you said

File: src/Tree.tsx (hunk 3 above)
Comment (shixin-guo) [reply src/Tree.tsx:35]: but I find,
```
const fileColors = { ...defaultFileColors, ...customizeFileColors };
```
this variable has existed, so I create a new one, so Is enough if we only change the user input (customize_file_colors => file_colors), and I guess the user does not care what name in our code

File: src/Tree.tsx [hunk 4]
```diff
@@ -58,9 +60,10 @@ const maxChildren = 9000;
 const lastCommitAccessor = (d) => new Date(d.commits?.[0]?.date + "0");
 const numberOfCommitsAccessor = (d) => d?.commits?.length || 0;
 export const Tree = (
-  { data, filesChanged = [], maxDepth = 9, colorEncoding = "type" }:
+  { data, filesChanged = [], maxDepth = 9, colorEncoding = "type", customizeFileColors}:
     Props,
 ) => {
+  const fileColors = { ...defaultFileColors, ...customizeFileColors };
```
Comment (shixin-guo) [inline src/Tree.tsx:66]: here

File: src/Tree.tsx (hunk 3 above)
Comment (shixin-guo) [reply src/Tree.tsx:35]: > let's mirror the change above - this should be `fileColors`

but I find,
```
const fileColors = { ...defaultFileColors, ...customizeFileColors };
```
this variable has existed, so I create a new one, Is it enough if we only change the user input (customize_file_colors => file_colors)? and I guess the user does not care what name in our code

Comment (Wattenberger) [reply src/Tree.tsx:35]: yep, makes sense! I like how you've changed things - one last change that might be helpful: do you mind changing `customizeFileColors` to `customFileColors` to be a bit more clear?

Comment (shixin-guo) [reply src/Tree.tsx]: No problem, done

**Review by Wattenberger — CHANGES_REQUESTED** [review]
Wonderful, thanks for that change! One last thing I forgot - we should add the input to the `action.yml` file:

https://github.com/githubocto/repo-visualizer/blob/main/action.yml#L5-L33

**Review by Wattenberger — APPROVED** [review]

**Comment by Wattenberger** [comment]
Looks wonderful, thanks for those changes!

**Pull request merged**